gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt

```diff
@@ -488,7 +537,12 @@ interface KspTask : Task {
 abstract class KspTaskJvm @Inject constructor(
     workerExecutor: WorkerExecutor,
     objectFactory: ObjectFactory
-) : KotlinCompile(KotlinJvmOptionsImpl(), workerExecutor, objectFactory), KspTask {
+) : KotlinCompile(
+        objectFactory.newInstance(KotlinJvmCompilerOptionsDefault::class.java),
```

Comment:
You can rely on injection so as to avoid referencing the KGP internal `KotlinJvmCompilerOptionsDefault`. Example: https://github.com/ting-yuan/ksp/blob/gradle/gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/tasks/inherited/InheritedTasks.kt#L268

Reply:
putting it in a constructor seems to expose internal classes (`KotlinJvmCompilerOptionsDefault` here) therefore is causing error.